test/test-shim.ts

```diff
@@ -70,6 +74,22 @@ test('modules already in cached do not show up in trace', t => {
   t.end();
 });
 
+test('preload traces the entire main execution and writes it to a file', t => {
+  const script = './build/test/fixtures/modA.js';
```

Comment:
This hard coded path (and others below) have a few problems:
* It assumes UNIX directory separators, so it won't work on windows.
* It assumes that the test will always be run from the root directory. If the user happens to be in a different directory it will fail.

My suggestion would be to use a path relative to `__dirname`.